### ISSUE TITLE
Remove data volume used for storing Che master logs and agents binaries 

### ISSUE THREAD

**Issue by garagatyi** [redhat-developer/rh-che#617]
**Issue problem:** 
There is an option not to use a volume for storing logs and agents binaries in [upstream Che](https://github.com/eclipse/che/pull/9236).
We should do it too [in rh-che](https://github.com/redhat-developer/rh-che/blob/rh-che6/openshift/rh-che.app.yaml#L33).

**Red Hat Che version:** 

version: (help/about menu)

 - [ ] I can reproduce it on latest official image

**Reproduction Steps:**

Describe how to reproduce the problem

**Runtime:**

runtime used:
   - [ ] minishift (include output of `minishift version`)
   - [ ] OpenShift.io
   - [ ] Openshift Container Platform (include output of `oc version`)
   - [ ] k8s (include output of `kubectl version`)

**Comment by garagatyi** [comment]
@l0rd I have discussed recreate strategy usage in Che 6.4.0 with upstream Che team and figured out that it won't help us in 6.4.0. Current upstream Che **does** stop workspaces on the stop of Che container. Functionality that changes it to workspace recovery is in branch for rolling update, not in the master branch. 
Because of this, we won't really benefit from fixing this issue in the current sprint. 
We can:
1. Set priority to P2 because of the reason I described above
1. Ask Che platform team to add this feature in Che 6.5.0 or 6.4.1

WDYT?
